src/client/camera.cpp

```diff
@@ -597,7 +597,11 @@ void Camera::update(LocalPlayer* player, f32 frametime, f32 tool_reload_ratio)
 
 void Camera::updateViewingRange()
 {
-	f32 viewing_range = g_settings->getFloat("viewing_range");
+	LocalPlayer *player = m_client->getEnv().getLocalPlayer();
+	assert(player);
+
+	PlayerFogSpec fog_spec = player->getFog();
+	f32 viewing_range = fog_spec.distance > 0.f ? fog_spec.distance : g_settings->getFloat("viewing_range");
```

Comment:
We should range-limit here as well, right? Client might change the setting after last distance was set from the server.